src/commonMain/kotlin/com/hoc081098/flowext/eager.kt

```diff
@@ -0,0 +1,158 @@
+ * @param bufferSize hints about the number of expected values from each inner [Flow]. Should be either
+ * a positive channel capacity or one of the constants defined in [Channel]. If inner [Flow] tries to emit
+ * more than [bufferSize] values before being concatenated to output, then it will be suspended
+ */
+@FlowPreview
+public fun <T, R> Flow<T>.flatMapConcatEager(
+  concurrency: Int = DEFAULT_CONCURRENCY,
+  bufferSize: Int = Channel.BUFFERED,
+  transform: suspend (value: T) -> Flow<R>,
+): Flow<R> = map(transform).flattenConcatEager(concurrency, bufferSize)
+
+/**
+ * Flattens the given flow of flows into a single flow in a sequential manner, without interleaving nested flows.
+ * But unlike [flattenConcat] collecting nested flows performed concurrently with a given [concurrency] limit
+ * on the number of concurrently collected flows.
+ *
+ * If [concurrency] is more than 1, then inner flows are collected by this operator *concurrently*.
+ * With `concurrency == 1` this operator is identical to [flattenConcat].
+ *
+ * ### Operator fusion
+ *
+ * Applications of [flowOn], [buffer], and [produceIn] _after_ this operator are fused with
+ * its concurrent merging so that only one properly configured channel is used for execution of merging logic.
+ *
+ * When [concurrency] is greater than 1, this operator is [buffered][buffer] by default
+ * and size of its output buffer can be changed by applying subsequent [buffer] operator.
+ *
+ * @param concurrency controls the number of in-flight flows, at most [concurrency] flows are collected
+ * at the same time. By default, it is equal to [DEFAULT_CONCURRENCY].
+ * @param bufferSize hints about the number of expected values from each inner [Flow]. Should be either
+ * a positive channel capacity or one of the constants defined in [Channel]. If [Flow] tries to emit
+ * more than [bufferSize] values before being concatenated to output, then it will be suspended
+ */
+@FlowPreview
+public fun <T> Flow<Flow<T>>.flattenConcatEager(
+  concurrency: Int = DEFAULT_CONCURRENCY,
+  bufferSize: Int = Channel.BUFFERED,
+): Flow<T> {
+  require(concurrency > 0) { "Expected positive concurrency level, but had $concurrency" }
+  return if (concurrency == 1) flattenConcat() else flattenConcatEagerInternal(this, concurrency, bufferSize)
+}
+
+private fun <T> flattenConcatEagerInternal(flow: Flow<Flow<T>>, concurrency: Int, bufferSize: Int) = channelFlow {
+  val collectors = Channel<ProxyCollector<T>>(concurrency - 1)
+  val job: Job? = coroutineContext[Job]
+  launch {
+    job?.ensureActive()
+    for (collector in collectors) {
+      collector.collectTo(this@channelFlow)
+    }
+  }
+  flow
+    .onCompletion { collectors.close() }
+    .collect { inner ->
+      job?.ensureActive()
+      val collector = ProxyCollector(Channel<T>(bufferSize))
+      collectors.send(collector)
+      launch {
+        try {
+          inner.collect(collector)
+        } finally {
+          collector.close()
+        }
```

Comment:
```suggestion
        }
        // All inner Flows were completed.
        if (collectors.isEmpty) {
          collectors.close()
        }
```
I think we can check `All inner Flows were completed.` via `Channel.isEmpty`